jointContribution/CHGNet/model/model.py

```diff
@@ -0,0 +1,669 @@
+        atom_feas = self.atom_embedding(g.atomic_numbers - 1)
+        bond_feas = self.bond_embedding(g.bond_bases_ag)
+        bond_weights_ag = self.bond_weights_ag(g.bond_bases_ag)
+        bond_weights_bg = self.bond_weights_bg(g.bond_bases_bg)
+        if len(g.angle_bases) != 0:
+            angle_feas = self.angle_embedding(g.angle_bases)
+        for idx, (atom_layer, bond_layer, angle_layer) in enumerate(zip(
+            self.atom_conv_layers[:-1], self.bond_conv_layers, self.
+            angle_layers, strict=False)):
+            atom_feas = atom_layer(atom_feas=atom_feas, bond_feas=bond_feas,
+                bond_weights=bond_weights_ag, atom_graph=g.
+                batched_atom_graph, directed2undirected=g.directed2undirected)
+            if len(g.angle_bases) != 0 and bond_layer is not None:
+                bond_feas = bond_layer(atom_feas=atom_feas, bond_feas=
+                    bond_feas, bond_weights=bond_weights_bg, angle_feas=
+                    angle_feas, bond_graph=g.batched_bond_graph)
+                if angle_layer is not None:
+                    angle_feas = angle_layer(atom_feas=atom_feas, bond_feas
+                        =bond_feas, angle_feas=angle_feas, bond_graph=g.
+                        batched_bond_graph)
+            if idx == self.n_conv - 2:
+                if return_atom_feas:
+                    prediction['atom_fea'] = paddle.split(x=atom_feas,
+                        num_or_sections=atoms_per_graph.tolist())
+                if compute_magmom:
+                    magmom = paddle.abs(x=self.site_wise(atom_feas))
+                    prediction['m'] = list(paddle.split(x=magmom.view(-
+                        1), num_or_sections=atoms_per_graph.tolist()))
+        atom_feas = self.atom_conv_layers[-1](atom_feas=atom_feas,
+            bond_feas=bond_feas, bond_weights=bond_weights_ag, atom_graph=g
+            .batched_atom_graph, directed2undirected=g.directed2undirected)
+        if self.readout_norm is not None:
+            atom_feas = self.readout_norm(atom_feas)
+        if self.mlp_first:
+            energies = self.mlp(atom_feas)
+            energy = self.pooling(energies, g.atom_owners).view(-1)
+            if return_site_energies:
+                prediction['site_energies'] = paddle.split(x=energies.
+                    squeeze(axis=1), num_or_sections=atoms_per_graph.tolist())
+            if return_crystal_feas:
+                prediction['crystal_fea'] = self.pooling(atom_feas, g.
+                    atom_owners)
+        else:
+            crystal_feas = self.pooling(atom_feas, g.atom_owners)
+            energy = self.mlp(crystal_feas).view(-1) * atoms_per_graph
+            if return_crystal_feas:
+                prediction['crystal_fea'] = crystal_feas
+        if compute_force:
+            force = paddle.grad(outputs=energy.sum(), inputs=g.
+                atom_positions, create_graph=True, retain_graph=True)
```

Comment:
此复现仅实现推理功能，把create_graph设置为False
```suggestion
                atom_positions, create_graph=False, retain_graph=True)
```